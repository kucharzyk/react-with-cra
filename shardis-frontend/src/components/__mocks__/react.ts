import * as React from 'react';

// Temporary hack to suppress error
// https://github.com/facebookincubator/create-react-app/issues/3199
window.requestAnimationFrame = function (callback) {
  setTimeout(callback, 0);
  return 0;
};

module.exports = React;
