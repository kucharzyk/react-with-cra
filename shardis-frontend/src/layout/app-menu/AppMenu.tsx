import * as React from 'react';

import { Icon, Menu } from 'antd';
import { Link } from 'react-router-dom';
import { RouteComponentProps, withRouter } from 'react-router';

class AppMenu extends React.Component<RouteComponentProps<{}  >> {
  render() {
    return (
      <Menu mode="horizontal" defaultSelectedKeys={[this.props.location.pathname]}>
        <Menu.Item key="/">
          <Link to={'/'}><Icon type="app"/>Main Page</Link>
        </Menu.Item>
        <Menu.Item key="/home">
          <Link to={'/home'}><Icon type="app"/>Home</Link>
        </Menu.Item>
        <Menu.Item key="/about">
          <Link to={'/about'}><Icon type="app"/>About</Link>
        </Menu.Item>
        <Menu.Item key="/time">
          <Link to={'/time'}><Icon type="app"/>Time</Link>
        </Menu.Item>
        <Menu.Item key="/404">
          <Link to={'/404'}><Icon type="app"/>404</Link>
        </Menu.Item>
      </Menu>
    );
  }
}

export default withRouter(AppMenu);
