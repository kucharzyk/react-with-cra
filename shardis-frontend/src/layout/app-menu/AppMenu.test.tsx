import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { create } from 'react-test-renderer';
import { StaticRouter } from 'react-router';
import AppMenu from './AppMenu';

describe('AppMenu', () => {

  const AppMenuWithRouter = () => (
    <StaticRouter location={'/'} context={{}}>
      <AppMenu/>
    </StaticRouter>
  );

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AppMenuWithRouter/>, div);
  });

  it('renders correctly', () => {
    const tree = create(<AppMenuWithRouter/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
