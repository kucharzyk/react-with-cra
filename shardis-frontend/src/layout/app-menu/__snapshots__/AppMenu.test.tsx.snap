// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppMenu renders correctly 1`] = `
<ul
  aria-activedescendant=""
  className="ant-menu ant-menu-light ant-menu-root ant-menu-horizontal"
  onKeyDown={[Function]}
  role="menu"
  style={Object {}}
  tabIndex="0"
>
  <li
    aria-disabled={undefined}
    aria-selected={true}
    className="ant-menu-item ant-menu-item-selected"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={Object {}}
    title={undefined}
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <i
        className="anticon anticon-app"
      />
      Main Page
    </a>
  </li>
  <li
    aria-disabled={undefined}
    aria-selected={false}
    className="ant-menu-item"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={Object {}}
    title={undefined}
  >
    <a
      href="/home"
      onClick={[Function]}
    >
      <i
        className="anticon anticon-app"
      />
      Home
    </a>
  </li>
  <li
    aria-disabled={undefined}
    aria-selected={false}
    className="ant-menu-item"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={Object {}}
    title={undefined}
  >
    <a
      href="/about"
      onClick={[Function]}
    >
      <i
        className="anticon anticon-app"
      />
      About
    </a>
  </li>
  <li
    aria-disabled={undefined}
    aria-selected={false}
    className="ant-menu-item"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={Object {}}
    title={undefined}
  >
    <a
      href="/time"
      onClick={[Function]}
    >
      <i
        className="anticon anticon-app"
      />
      Time
    </a>
  </li>
  <li
    aria-disabled={undefined}
    aria-selected={false}
    className="ant-menu-item"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={Object {}}
    title={undefined}
  >
    <a
      href="/404"
      onClick={[Function]}
    >
      <i
        className="anticon anticon-app"
      />
      404
    </a>
  </li>
</ul>
`;
